src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -6577,11 +6630,13 @@ public Expression SetValue(Compiler compiler, Expression rhs)
             var args = GetArgumentExprs(compiler);
             if (memberNameAst != null)
             {
-                return compiler.InvokeMember(memberNameAst.Value, constraints, target, args.Append(rhs), @static: false, propertySet: true);
+                var expr = compiler.InvokeMember(memberNameAst.Value, constraints, target, args.Append(rhs), @static: false, propertySet: true);
+                return InvokeMemberExpressionAst.NullConditionalAccess ? Compiler.GetNullCheckExpressionBlock(target, expr) : expr;
             }
 
             var memberNameExpr = GetMemberNameExpr(compiler);
-            return compiler.InvokeDynamicMember(memberNameExpr, constraints, target, args.Append(rhs), @static: false, propertySet: true);
+            var exprNullName = compiler.InvokeDynamicMember(memberNameExpr, constraints, target, args.Append(rhs), @static: false, propertySet: true);
+            return InvokeMemberExpressionAst.NullConditionalAccess ? Compiler.GetNullCheckExpressionBlock(target, exprNullName) : exprNullName;
```

Comment:
Similarly, assignment to InvokeMemberExpression should be banned at the semantic check step.

Reply:
Removed